### DISCUSSION TITLE
[docs][guides] Personalization in Gatsby

### DISCUSSION THREAD

**Discussion by marcysutton** [gatsbyjs/gatsby#31219] [Ideas / Feature Requests]
## Summary

At the Gatsby Discovery Camp, we had a great session on Personalization and A/B testing. These workflows are important enough to folks that there have been related GitHub issues and it was brought up as an unconference session–for those reasons they should be properly documented. 

We should document the multiple techniques available here, and let developers choose what works for them. This page makes sense to live under [Adding Website Functionality](https://www.gatsbyjs.org/docs/adding-website-functionality/) in the reference guides section of the docs.

Notes from the Unconference session, including recommended services/techniques: https://docs.google.com/document/d/1X-oKnmkGp5M-Klw7mFnR257cU83DgjT0KjgF1sEEFNI/edit#

## Related issues
- "Document A/B testing" https://github.com/gatsbyjs/gatsby/issues/5936
- "Disabling cache-busting JS URLs" https://github.com/gatsbyjs/gatsby/issues/5262

Please refer to the [Gatsby Style Guide](https://www.gatsbyjs.org/contributing/gatsby-style-guide/) and [Reference Guide template](https://www.gatsbyjs.org/contributing/docs-templates/#reference-guides) when creating this document.

cc @jaredsilver

**Comment by jaredsilver** [comment]
@gatsbot Not stale

**Comment by gabrielgene** [comment]
Hey @marcysutton 
I think I can do that.
Is all the documentation in this document? https://docs.google.com/document/d/1X-oKnmkGp5M-Klw7mFnR257cU83DgjT0KjgF1sEEFNI/edit#

**Comment by marcysutton** [comment]
@gabrielgene this one was actually assigned on Friday. But if anything changes, we'll let you know!

**Comment by stoltzrobin** [comment]
@marcysutton I read the documentation that @gabrielgene linked. I've been in a similar situation where I needed A/B-testing and used Gatsby. I couldn't use frameworks that uses client-side A/B test due to speed requirements and initial loading times was to slow. 

What I've used is a solution where I load css high up in the body so that the user will not see any flickering, and it's also possible to use react component to render above the fold without any flickering. It adds nods to the DOM-tree but makes it also possible to build a A/B-testing solution without the need of splitting traffic at the CDN. The only thing needed is an analytic framework to send the data to.

**Comment by marcysutton** [comment]
@stoltzrobin circling back to this, that sounds super interesting! I'd love to hear more about your approach: would you be interested in writing about it? We just published a guide on A/B testing with Google Analytics and Netlify, and would love to include other techniques as part of the section: https://www.gatsbyjs.org/docs/ab-testing/

**Comment by BraunreutherA** [comment]
Hi @stoltzrobin and @marcysutton I've been working for nearly a year now on personalization on the edge with cloudflare workers and am applying personalized content on the fly also to gatsby sites. The system works actually similar to what you describe in the documentation you wrote. 
I'm thinking to go the next step and provide headless personalization as a Service in the CDN / Edge level for JAMstack sites. Would this be a topic where you're interested to chat about? I'd also like to provide a plugin and work with gatsby as one of the first integrations as I use it a lot myself.
